app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -0,0 +1,422 @@
+  void fetchKnownSubChatsData() async {
+    final selectedParentSpaceId = ref.watch(selectedSpaceIdProvider);
+    if (selectedParentSpaceId == null) return;
+    final space = await ref
+        .watch(spaceRelationsOverviewProvider(selectedParentSpaceId).future);
+
+    childRoomsIds.clear();
+    if (childRoomType == ChildRoomType.chat) {
+      for (int i = 0; i < space.knownChats.length; i++) {
+        childRoomsIds.add(space.knownChats[i].getRoomId().toString());
+      }
+    } else {
+      for (int i = 0; i < space.knownSubspaces.length; i++) {
+        childRoomsIds.add(space.knownSubspaces[i].getRoomId().toString());
+      }
+      //Add recommended child spaces ids
+      recommendedChildSpaceIds.clear();
+      for (int i = 0; i < space.otherRelations.length; i++) {
+        recommendedChildSpaceIds
+            .add(space.otherRelations[i].getRoomId().toString());
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+
+    return SideSheet(
+      header: sheetHeader,
+      body: SizedBox(
+        height: size.height - 120,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            parentSpaceDataUI(),
+            searchUI(),
+            Expanded(child: roomList()),
+          ],
+        ),
+      ),
+    );
+  }
+
+  //Search
+  Widget searchUI() {
+    return Search(
+      onChanged: (value) {
+        if (childRoomType == ChildRoomType.chat) {
+          ref.read(chatSearchValueProvider.notifier).update((state) => value);
+        } else if (childRoomType == ChildRoomType.space) {
+          ref.read(spaceSearchValueProvider.notifier).update((state) => value);
+        }
+      },
+      searchController: searchTextEditingController,
+    );
+  }
+
+  //Parent space
+  Widget parentSpaceDataUI() {
+    final spaceDetails = ref.watch(selectedSpaceDetailsProvider);
+    return Padding(
+      padding: const EdgeInsets.all(12),
+      child: spaceDetails.when(
+        data: (space) {
+          return space == null
+              ? const SizedBox.shrink()
+              : Column(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    const Text('Parent Space'),
+                    SpaceChip(space: space),
+                  ],
+                );
+        },
+        error: (e, s) => errorUI('error: ', e),
+        loading: () => Container(),
+      ),
+    );
+  }
+
+  //Manage list data based on the child room type
+  Widget roomList() {
+    if (childRoomType == ChildRoomType.chat) {
+      return chatsList();
+    } else if (childRoomType == ChildRoomType.space ||
+        childRoomType == ChildRoomType.recommendedSpace) {
+      return spacesList();
+    } else {
+      return const SizedBox.shrink();
+    }
```

Comment:
No other cases exist, do they?

```suggestion
    if (widget.childRoomType == ChildRoomType.chat) {
      return chatsList();
    } else {
      return spacesList();
    }
```